src/cli/subcommand/restore_shell.rs

```diff
@@ -0,0 +1,80 @@
+use std::{os::unix::prelude::PermissionsExt, process::ExitCode};
+
+use crate::{
+    action::common::ConfigureShellProfile,
+    cli::{ensure_root, CommandExecute},
+    planner::{PlannerError, ShellProfileLocations},
+};
+use clap::{ArgAction, Parser};
+
+/**
+Update the macOS startup files to make Nix usable after system upgrades.
```

Comment:
I'm under the impression this can be used on non-Macs too? (I don't see it `cfg`'d off by target) If it works on Linux I'd like to make it available for every body! =D